### ISSUE TITLE
Spike: Drupal JSON:API UI/explorer

### ISSUE THREAD

**Issue by timcosgrove** [department-of-veterans-affairs/va.gov-cms#9135]
## Description
We need a way to visualize and explore Drupal's JSON:API so that developers interacting with the CMS can understand what the CMS data structures are and how they can make requests for data.

## Acceptance Criteria
- [x] JSON:API Visualization options are explored, and featured-compared
- [x] Any issues with each are documented
- [x] If there's a clear choice, go ahead and install

### Drupal JSON:API
Both Redoc and Swagger UI are provided as possible UI for Drupal. See https://www.drupal.org/project/openapi_ui

Another option: https://www.drupal.org/project/jsonapi_explorer

These UI libraries have a fair amount of trouble with our data model, most likely due to its size. Whatever solution we come up with will likely have to account for breaking up how much of the API structure is shown at once.

We should also consider what pieces of the API response we may want to filter away.

If there are options for query builders that can be explored, that's also a good options.

### CMS Team
Please check the team(s) that will do this work.

- [x] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by TheBoatyMcBoatFace** [comment]
@timcosgrove is this basically: 
- #9497?

**Comment by alexfinnarn** [comment]
I'll try to create a quasi-rubric for evaluating the two options of Redoc and Swagger.

Swagger:

1. Installation - Swagger has a UI library that needs to be installed separately. I installed it manually for testing, but there are notes to add the library via Composer and require another dependency `mnsami/composer-custom-directory-installer` to make it work. 
2. Project Health - The library repo at https://github.com/swagger-api/swagger-ui/pulse/monthly has a decent amount of activity and looks maintained.
3. Initial Load - The whole JSON:API schema is able to load. TTI seemed like ~10 seconds and everything completes ~25 seconds. I think the particular route information is lazy loaded once you click to expand that route so it helps with loading so many resources.
4. Features - Swagger has a way to test authentication, which is nice. I didn't see a search but "ctl/cmd+F" works well enough since there's no pagination. You can hide/show the resources, and the experience is the same as when I've used Swagger before. It is easy enough to test out API calls and view the resource definitions as well as query parameters. 

Redoc:

1. Installation - Redoc has no external dependencies and is only one Drupal module to install.
2. Project Health - The library repo at https://github.com/Redocly/redoc/pulse/monthly has some activity but looks to be a smaller project than Swagger.
3. Initial Load - I saw a "loading" indicator ~11 seconds but I couldn't do anything until ~2.1 minutes had passed. 
5. Features - I couldn't use the UI. It crashed and said `Error code: RESULT_CODE_HUNG` which was new to me. 

Conclusions:

1. Installation - Redoc has fewer dependencies so I say it wins in the installation category. 
2. Project Health - Swagger wins since it appears to be more popular and therefore the greatest chance of longevity. 
3. Initial Load - Swagger definitely wins in this category.
4. Features - Swagger wins again since I could actually use it. I think the JSON:API schema would have to be modified to get Redoc to work, but Swagger responsiveness is okay without doing anything. I wouldn't be too miffed using it occasionally to see what the JSON:API had to offer. 

So in conclusion, the Swagger UI Library was the only option that worked for me. The only caveat there is how to load the library. I kind of like the simplicity of downloading the release and think we wouldn't update this dependency often, but I see that the libraries directory is git ignored. You could still keep a copy in the repo and then use composer scaffolding to copy it in, but I'm guessing the composer route will be wanted.

**Comment by alexfinnarn** [comment]
For adding Swagger UI, this patch apparently allows filtering: https://www.drupal.org/project/openapi_ui_swagger/issues/3259746 It's only one line of config. 

For brownie points, I could make it into a form variable and update the patch/MR so users can choose whether they want the filter. I'm betting there are other useful config options as well.

**Comment by tjheffner** [comment]
I think the load time alone is reason enough to move forward with Swagger here. It's also already used in a few places around the VA, so it's a familiar tool to many.

https://department-of-veterans-affairs.github.io/va-digital-services-platform-docs/api-reference/#/
https://github.com/department-of-veterans-affairs/abd-vro/wiki/Testing-using-Swagger-UI
etc.

Let's move forward with Swagger UI (filtering would be nice, whether that's surfaced to a config or not. config might be nice)

**Comment by alexfinnarn** [comment]
I tried for a little while not to install `mnsami/composer-custom-directory-installer` since there's at least one other custom directory installer already in `composer.json` but I couldn't get it to work without that plugin. Hopefully, it doesn't interfere with the others, but it'd be great for Composer to add this feature by default. I feel like a lot of projects need to install dependencies in other directories these days...oh well.

I also added a patch to the filtering d.o. issue, https://www.drupal.org/project/openapi_ui_swagger/issues/3259746#comment-15359039, and I included more config options. Deep linking and collapsing all the fields. 

The draft PR is building now, and I will wait to look at Tugboat before making it an official PR.

**Comment by alexfinnarn** [comment]
A follow-up issue: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16456
